examples/GettingStarted/MulticlassClassification_Iris/Program.cs

```diff
@@ -0,0 +1,123 @@
+                //              ColumnConcatenator transform:
+                new ColumnConcatenator("Features",
+                    "SepalLength",
+                    "SepalWidth",
+                    "PetalLength",
+                    "PetalWidth"),
+                // StochasticDualCoordinateAscentClassifier is an algorithm that will be used to train the model.
+                new StochasticDualCoordinateAscentClassifier()
+            };
+
+            Console.WriteLine("=============== Training model ===============");
+            // The pipeline is trained on the dataset that has been loaded and transformed.
+            var model = pipeline.Train<IrisData, IrisPrediction>();
+
+            // Saving the model as a .zip file.
+            await model.WriteAsync(ModelPath);
+
+            Console.WriteLine("=============== End training ===============");
+            Console.WriteLine("The model is saved to {0}", ModelPath);
+
+            return model;
+        }
+
+        private static void Evaluate(PredictionModel<IrisData, IrisPrediction> model)
+        {
+            // To evaluate how good the model predicts values, the model is ran against new set
+            // of data (test data) that was not involved in training.
+            var testData = new TextLoader<IrisData>(TestDataPath, useHeader: false);
+
+            // ClassificationEvaluator performs evaluation for Multiclass Classification type of ML problems.
+            var evaluator = new ClassificationEvaluator {OutputTopKAcc = 3};
+            
+            Console.WriteLine("=============== Evaluating model ===============");
+
+            var metrics = evaluator.Evaluate(model, testData);
+            Console.WriteLine("Metrics:");
+            Console.WriteLine($"    AccuracyMacro = {metrics.AccuracyMacro:0.####}, a value between 0 and 1, the closer to 1, the better");
+            Console.WriteLine($"    AccuracyMicro = {metrics.AccuracyMicro:0.####}, a value between 0 and 1, the closer to 1, the better");
+            Console.WriteLine($"    LogLoss = {metrics.LogLoss:0.####}, the closer to 0, the better");
+            Console.WriteLine($"    LogLoss for class 1 = {metrics.PerClassLogLoss[0]:0.####}, the closer to 0, the better");
+            Console.WriteLine($"    LogLoss for class 2 = {metrics.PerClassLogLoss[1]:0.####}, the closer to 0, the better");
+            Console.WriteLine($"    LogLoss for class 3 = {metrics.PerClassLogLoss[2]:0.####}, the closer to 0, the better");
+            Console.WriteLine();
+            Console.WriteLine($"    ConfusionMatrix:");
+
+            // Print confusion matrix
+            for (var i = 0; i < metrics.ConfusionMatrix.Order; i++)
+            {
+                for (var j = 0; j < metrics.ConfusionMatrix.ClassNames.Count; j++)
+                {
+                    Console.Write("\t" + metrics.ConfusionMatrix[i, j] + "\t");
```

Comment:
>Console.Write("\t" + metrics.ConfusionMatrix[i, j] + "\t"); [](start = 20, length = 59)

Were the double tabs on both the start *and* end intentional?

Reply:
It was intentional; it's using `Console.Write` not `WriteLine`, so the trailing tab is for subsequent writes.

Reply:
Are you *sure*? I could see having a trailing tab, or a leading tab, but I don't think having both is intentional?

Let's imagine you're outputting, on one row, the numbers, say, 0, 1, 2, 3. The output is `\t0\t\t1\t\t2\t\t3\t`. Right? That seems wrong.

---
In reply to: [190758071](https://github.com/dotnet/machinelearning/pull/238#discussion_r190758071) [](ancestors = 190758071)